File: my-solutions/0435-non-overlapping-intervals/solution.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& inl) {
        int n=inl.size();
        for(int i=0;i<n;i++){
            swap(inl[i][0],inl[i][1]);
        }
        sort(inl.begin(),inl.end());
        int c=1;
        int l=inl[0][0];
        for(int i=1;i<n;i++){
            if(inl[i][1]>=l){
                l=inl[i][0];
                c++;
            }
        }
        return n-c;
    }
};

File: my-solutions/0435-non-overlapping-intervals/solution.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key=lambda interval: interval[1])
        lastStart, lastEnd = intervals[0]
        rm = 0

        for i in range(1, len(intervals)):
            currStart, currEnd = intervals[i]
            if currStart < lastEnd:
                rm += 1
                lastEnd = min(lastEnd, currEnd)
            else:
                lastStart, lastEnd = currStart, currEnd
        
        return rm
